src/stores/TransitionState.ts

```diff
@@ -0,0 +1,117 @@
+import { RouterState } from './RouterState';
+import { RouterStore, Route } from './RouterStore';
+import { valueEqual } from '@react-force/utils';
+import Debug from 'debug';
+
+const debug = Debug('msr:RouterStore');
+
+export class TransitionState {
+    private transitionState?: RouterState;
+    private readonly fromRoute: Route;
+
+    constructor(
+        private routerStore: RouterStore,
+        private readonly fromState: RouterState
+    ) {
+        // Create an object copy so we can modify hooks during transition.
```

Comment:
There's a comment below to an issue where someone is using classes to implement their routes - wondering if this is compatible with that (https://github.com/nareshbhatia/mobx-state-router/issues/74).

Reply:
I don't see why not, but @nareshbhatia can confirm us.